aten/src/ATen/native/mkldnn/xpu/Conv.cpp

```diff
@@ -0,0 +1,726 @@
+  bool deterministic;
+
+  bool is_strided() const;
+  bool is_dilated() const;
+  bool is_padded() const;
+  bool is_output_padding_neg() const;
+  bool is_output_padding_big() const;
+  bool is_padding_neg() const;
+  bool is_stride_nonpos() const;
+  void view1d_as_2d();
+  bool use_cpu_depthwise3x3_winograd(
+      const at::Tensor& input,
+      const at::Tensor& weight) const;
+  bool is_depthwise(const at::Tensor& input, const at::Tensor& weight) const;
+};
+
+std::ostream& operator<<(std::ostream& out, const ConvParams& params) {
+  out << "ConvParams {"
+      << "  stride = " << IntArrayRef{params.stride}
+      << "  padding = " << IntArrayRef{params.padding}
+      << "  dilation = " << IntArrayRef{params.dilation}
+      << "  transposed = " << params.transposed
+      << "  output_padding = " << IntArrayRef{params.output_padding}
+      << "  groups = " << params.groups << "  benchmark = " << params.benchmark
+      << "  deterministic = " << params.deterministic << "}";
+  return out;
+}
+
+bool ConvParams::is_strided() const {
+  bool is_strided = false;
+  for (int s : stride) {
+    is_strided |= (s != 1);
+  }
+  return is_strided;
+}
+
+bool ConvParams::is_dilated() const {
+  bool is_dilated = false;
+  for (int d : dilation) {
+    is_dilated |= (d != 1);
+  }
+  return is_dilated;
+}
+
+bool ConvParams::is_padded() const {
+  bool is_padded = false;
+  for (int p : padding) {
+    is_padded |= (p != 0);
+  }
+  return is_padded;
+}
+
+bool ConvParams::is_output_padding_neg() const {
+  bool is_non_neg = false;
+  for (int p : output_padding) {
+    is_non_neg |= (p < 0);
+  }
+  return is_non_neg;
+}
+
+bool ConvParams::is_output_padding_big() const {
+  bool is_big = false;
+  for (size_t i = 0; i < output_padding.size(); i++) {
+    is_big |=
+        (output_padding[i] >= stride[i] || output_padding[i] >= dilation[i]);
+  }
+  return is_big;
+}
+
+bool ConvParams::is_padding_neg() const {
+  bool is_non_neg = false;
+  for (int p : padding) {
+    is_non_neg |= (p < 0);
+  }
+  return is_non_neg;
+}
+
+bool ConvParams::is_stride_nonpos() const {
+  bool is_nonpos = false;
+  for (int s : stride) {
+    is_nonpos |= (s <= 0);
+  }
+  return is_nonpos;
+}
+
+void ConvParams::view1d_as_2d() {
+  if (stride.size() == 1) {
+    stride.insert(stride.begin(), 1);
+    padding.insert(padding.begin(), 0);
+    dilation.insert(dilation.begin(), 1);
+    output_padding.insert(output_padding.begin(), 0);
+  }
+}
+
+bool ConvParams::use_cpu_depthwise3x3_winograd(
+    const at::Tensor& input,
+    const at::Tensor& weight) const {
+  return false;
+}
+
+bool ConvParams::is_depthwise(const at::Tensor& input, const at::Tensor& weight)
+    const {
+  return !transposed && input.ndimension() == 4 && input.size(1) == groups &&
+      groups > 1 && // no point if there is only a single group
+      weight.size(0) % input.size(1) ==
+      0; // output channels must be a multiple of input channels
+}
+
+static void check_shape_forward(
```

Comment:
ditto.

Reply:
Same as `ConvParams` above, `check_shape_forward` is a static function defined in `convolution.cpp`, it may not be the part of API of torch for we using it.